File: Coding Challenge 04/primes.rb
def primes(max)
    primes_arr = []
    
    i = 2
    while i < max
        if isprime(i)
            # i is prime; add it to the array
            primes_arr << i
        end
        
        i += 1
    end
    
    # return primes_arr
    primes_arr
end

# Works for values greater than 1

def isprime(num)
    i = 2
    while i < num
        is_divisible = ((num % i) == 0)
        
        if is_divisible
            # divisor found; stop and return false!
            return false
        end
        
        i += 1
    end
    
    # no divisors found
    true
end

File: Coding Challenge 04/primes.m
function result=primes(max)
    result=[];
    for i=2:max
        if isprime(i)
            result=[result i];
        end
    end
end

%works for values greater than 1
function result=isprime(num)
    for i=2:num-1
        if rem(num,i)==0
            result=false;
            return;
        end
    end
    result=true;
end